modules/gdscript/gdscript_cache.cpp

```diff
@@ -158,6 +158,10 @@ void GDScriptCache::remove_script(const String &p_path) {
 
 	MutexLock lock(singleton->mutex);
 
+	if (singleton->destructing) {
+		return;
+	}
+
```

Comment:
For some reason, without this, the tests will result in a `heap-use-after-free` error with the address sanitizer.

Reply:
That is so confusing! I don't see why these changes would do that.

Could it be a regression from elsewhere that shows up for you if you've rebased from master?

Reply:
i'm suspicious of this part, but the rest looks great!

Reply:
I'll create a new PR for that part, it doesn't belong in this PR.